akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala

```diff
@@ -87,8 +87,6 @@ private[akka] class InputStreamPublisher(is: InputStream, bytesReadPromise: Prom
 
   /** BLOCKING I/O READ */
   def loadChunk() = try {
-    val arr = Array.ofDim[Byte](chunkSize)
```

Comment:
ok, I didn't see that ByteString1C was used. Isn't then simple to just use ByteString.apply on the shared array, which will do the proper copying in turn.